classes/services/PKPSubmissionService.inc.php

```diff
@@ -795,6 +795,12 @@ public function delete($submission) {
 	public function canEditPublication($submissionId, $userId) {
 		$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO'); /* @var $stageAssignmentDao StageAssignmentDAO */
 		$stageAssignments = $stageAssignmentDao->getBySubmissionAndUserIdAndStageId($submissionId, $userId, null)->toArray();
+		$submission = $this->get($submissionId);
+                $userIsAuthor = !empty($stageAssignmentDao->getBySubmissionAndRoleId($submissionId, ROLE_ID_AUTHOR, null, $userId)->toArray());
+                // If the submission is declined and the current user is an author of the submission
+                if ($submission->getStatus() == STATUS_DECLINED && $userIsAuthor) {
+                        return false;
```

Comment:
I think a more complex policy will be required. If someone is an editor, for example, they probably should have permission to edit metadata on declined submissions they also authored. This restriction should probably apply to submissions only when the author is the only stage assignment.

Reply:
> I think a more complex policy will be required. If someone is an editor, for example, they probably should have permission to edit metadata on declined submissions they also authored.

I [added a check](https://github.com/pkp/pkp-lib/pull/10693/commits/e49e800c7d3189aaf97680d724de8f67ebd3cf33) if the user doesn't have permissions to edit metadata according to the role.

Reply:
> This restriction should probably apply to submissions only when the author is the only stage assignment.

Sorry, I didn't understand exactly what you meant. Is that a necessary validation in [this conditional](https://github.com/pkp/pkp-lib/blob/e49e800c7d3189aaf97680d724de8f67ebd3cf33/classes/services/PKPSubmissionService.inc.php#L801), too?

Reply:
> Sorry, I didn't understand exactly what you meant. Is that a necessary validation in [this conditional](https://github.com/pkp/pkp-lib/blob/e49e800c7d3189aaf97680d724de8f67ebd3cf33/classes/services/PKPSubmissionService.inc.php#L801), too?

I discussed it internally with others and understood the point. In the current PR code, there are still issues, such as an author having the role of Moderator in the OPS but not being able to edit the declined submission. I will make the necessary adjustments.

Reply:
I've [adjusted](https://github.com/pkp/pkp-lib/pull/10693/commits/2819c8ed645cbfcd4e54df42e987fd059bc5fb55) the restriction to apply only in cases where the author has no role other than author/reader.